packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -757,3 +760,79 @@ defineQuery({
     return { tags, totalCount };
   },
 });
+
+defineMutation({
+  name: "promoteLensToMain",
+  resultType: "Boolean",
+  argTypes: "(lensId: String!)",
+  fn: async (root, args: {lensId: string}, context) => {
+    const { lensId } = args;
+    if (!userIsAdminOrMod(context.currentUser)) {
+      throw new Error("Only admins can promote lenses to main");
+    }
+
+    // eslint-disable-next-line no-console
+    console.log(`Promoting lens ${lensId} to main`);
+    
+    // Find the lens and corresponding tag
+    const lensMultiDocument = await context.loaders.MultiDocuments.load(lensId);
+    if (!lensMultiDocument) throw new Error("Lens ID not found: "+lensId);
+    const tagId = lensMultiDocument.parentDocumentId;
+    const tag = await context.loaders.Tags.load(tagId);
+    if (!tag) throw new Error("Could not find corresponding tag for lens ID: "+lensId);
+    
+    // Swap revisions
+    const tagRevisions = await Revisions.find({ documentId: tagId }).fetch();
+    const lensRevisions = await Revisions.find({ documentId: lensId }).fetch();
+    const latestTagRevisionId = tag.description_latest;
+    const latestLensRevisionId = lensMultiDocument.contents_latest;
+    console.log(`Moving ${tagRevisions.length} revisions from tag to lens`); //eslint-disable-line no-console
+    await Revisions.rawUpdateMany(
```

Comment:
I guess if we're not doing these 4 updates in a transaction, then having them happen sequentially probably makes sense, since it makes it easier to figure out recovery from failures